kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncProcessor.java

```diff
@@ -69,12 +66,18 @@
  * -Coordinates the handoff of records between the StreamThread and AyncThreads
  * -The starting and ending point of all async events -- see {@link AsyncEvent}
  */
-public class AsyncProcessor<KIn, VIn, KOut, VOut>
-    implements Processor<KIn, VIn, KOut, VOut>, FixedKeyProcessor<KIn, VIn, VOut> {
+public class AsyncProcessor<KIn, VIn, KOut, VOut> {
 
-  // Exactly one of these is non-null and the other is null
-  private final Processor<KIn, VIn, KOut, VOut> userProcessor;
-  private final FixedKeyProcessor<KIn, VIn, VOut> userFixedKeyProcessor;
+  private final String logPrefix;
+  private final Logger log;
+
+  private final String streamThreadName;
+  private final String asyncProcessorName;
+  private final TaskId taskId;
+
+  private final AsyncThreadPool threadPool;
+  private final SchedulingQueue<KIn> schedulingQueue;
+  private final FinalizingQueue finalizingQueue;
```

Comment:
The nice thing about this change is that we only instantiate the AsyncProcessor after the processor is initialized, so we can actually make all these variables final. I was hoping to do this anyway